.github/container/pip-finalize.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -ex -o pipefail
+
+pushd /opt/pip-tools.d
+
+pip-compile -o requirements.txt $(ls manifest.*)
```

Comment:
This still yields unpinned VCS installs. I believe we still need a mechanism like this: https://github.com/NVIDIA/JAX-Toolbox/pull/271/files#diff-8386cfe5c29a3be212a911c9550a56c57420a17cae190b80f60e52c0ce265645R151-R175

Reply:
Does this actually pin the versions or only document the SHA of the VCS installs at the time of pip-compile?

Reply:
IIRC this will pin it

Reply:
After speaking with @DwarKapex , we will just address this in another PR since it's better to get this out ASAP. I will file an issue for this